### PR TITLE
Backports 0.19 pr11

### PR DESCRIPTION
**Pull request by Munkybooty**
I'm unable to build so this may be my worst pr yet. apologies in advance for that if there is a lot of trivial errors that have been missed.

### CODE REVIEW

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: test/functional/feature_shutdown.py [hunk 1]
```diff
@@ -20,8 +20,14 @@ def set_test_params(self):
 
     def run_test(self):
         node = get_rpc_proxy(self.nodes[0].url, 1, timeout=600, coveragedir=self.nodes[0].coverage_dir)
+        # Force connection establishment by executing a dummy command.
+        node.getblockcount()
         Thread(target=test_long_call, args=(node,)).start()
-        # wait 1 second to ensure event loop waits for current connections to close
+        # Wait until the server is executing the above `waitfornewblock`.
+        wait_until(lambda: len(self.nodes[0].getrpcinfo()['active_commands']) == 2)
```
Comment (UdjinM6) [inline test/functional/feature_shutdown.py:27]: 14958: requires 14982

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: src/rpc/server.cpp [hunk 2]
```diff
@@ -191,11 +220,37 @@ static UniValue uptime(const JSONRPCRequest& jsonRequest)
     return GetTime() - GetStartupTime();
 }
 
+static UniValue getrpcinfo(const JSONRPCRequest& request)
+{
+    if (request.fHelp || request.params.size() > 0) {
+        throw std::runtime_error(
+            RPCHelpMan{"getrpcinfo",
+                "\nReturns details of the RPC server.\n", {}}
+                .ToString()
```
Comment (UdjinM6) [inline src/rpc/server.cpp:229]: 14982: add missing parts from 14987 we backported earlier
```suggestion
                "\nReturns details of the RPC server.\n",
                {},
                RPCResults{},
                RPCExamples{""},
            }.ToString()
```

File: doc/release-notes-15637.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+RPC changes
+-----------
+In getmempoolancestors, getmempooldescendants, getmempoolentry and getrawmempool RPCs, to be consistent with the returned value and other RPCs such as getrawtransaction, vsize has been added and size is now deprecated. size will only be returned if bitcoind is started with `-deprecatedrpc=size`.
```
Comment (UdjinM6) [inline doc/release-notes-15637.md:3]: 15637: this backport is also a part of #4609

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

**Review by UdjinM6 — APPROVED** [review]
utACK

**Review by PastaPastaPasta — APPROVED** [review]
utACK for merging via merge commit

**Pull request merged**